steam_community_market/market.py

```diff
@@ -1,84 +1,66 @@
-    :raises SteamUnsupportedCurrency: Raised when a currency is not supported by the Steam Community Market anymore.
+    It allows users to get price and volume information for items in the Steam Community Market and supports multiple currencies.
     """
 
     URI = "http://steamcommunity.com/market/priceoverview"
 
     def __init__(
-        self, currency: int | str | ESteamCurrency = ESteamCurrency.USD
+        self,
+        currency: Union[
+            int, str, SteamCurrency, SteamLegacyCurrency
+        ] = SteamCurrency.USD,
     ) -> None:
-        """Sets the currency to be outputted.
+        """Initializes the Market object with a specified currency.
 
         :param currency: Currency used for prices.
-        :type currency: :class:`int`, :class:`str`, :class:`ESteamCurrency`
-        :raises SteamUnsupportedCurrency: Raised when a currency is not supported by the Steam Community Market anymore.
+        :type currency: int or str or SteamCurrency or SteamLegacyCurrency
+        :value currency: SteamCurrency.USD
+        :raises InvalidCurrencyException: Raised when a currency is considered to be invalid by the Steam Community Market.
+        :raises LegacyCurrencyException: Raised when a currency is not supported by the Steam Community Market anymore.
         """
-        if isinstance(currency, ESteamUnsupportedCurrency):
-            raise SteamUnsupportedCurrency(currency)
 
         self.currency = self._supported_currency(currency)
 
-    @staticmethod
-    def _supported_currency(
-        currency: int | str | ESteamCurrency,
-    ) -> ESteamSupportedCurrency:
-        """Returns a supported currency.
-
-        :param currency: Currency used for prices.
-        :type currency: :class:`int`, :class:`str`, :class:`ESteamCurrency`
-        :return: A supported currency.
-        :rtype: :class:`ESteamSupportedCurrency`
-
-        .. versionadded:: 1.2.4
-        """
-
-        if isinstance(currency, int):
-            return ESteamSupportedCurrency(currency)
-        if isinstance(currency, str):
-            return ESteamSupportedCurrency[currency.upper()]
-        if isinstance(currency, ESteamSupportedCurrency):
-            return currency
+    def get_overview(
+        self, name: str, app_id: Union[int, AppID]
+    ) -> Optional[dict[str, Union[bool, str]]]:
+        """Gets the prices and volume of an item in the Steam Community Market.
 
-        return ESteamSupportedCurrency.USD
-
-    def get_overview(self, name: str, app_id: int | AppID) -> dict | None:
-        """Gets the prices and volume of an item.
+        .. versionadded:: 1.0.0
+        .. versionchanged:: 1.3.0
+            * Moved :param:`app_id`'s validation to :meth:`_app_id_value`.
+            * Improved type hinting for both parameters and return value.
+            * Removed :param:`name`'s validation check and just called :meth:`_fix_item_name`.
+            * Simplified the return statement.

```

Comment:
Typically since the end user doesn't care about changes like this they shouldn't be included. If you want to have a history of these kinds of changes use better git commit titles

Reply:
I thought it would be nice to have a changelog for the latest `.. versionchanged::` directive.
For example, if the above method would change in `1.3.2`, the changelog related to `1.3.0` will get removed and replaced by the changelog of `1.3.2`.